plugins/common/forward/src/index.ts

```diff
@@ -111,50 +161,139 @@ export function apply(ctx: Context, { rules, interval }: Config) {
+      authority: 3,
+      checkArgCount: true,
     })
+      .option('user', '-U <user:user>')
+      .option('flag', '-F <flag:string>')
+      .option('all', '-A')
+      .check(({
+        session,
+        options,
+      }) => {
+        if (JSON.stringify(options) === '{}') options.all = true
+      })
+      .action(async ({
+        session,
+        options,
+      }, channel) => {
+        try {
+          await ctx.database.create('forward', {
+            source: session.cid,
+            target: channel,
+            filter: generateFilter(options),
+          })
+          session.text('.changed')
+        } catch (e) {
+          ctx.logger('forward').error(e)
+        }
+      })
 
-    register('.remove <channel:channel>', async ({ session }, id) => {
-      const { forward } = session.channel
-      const index = forward.indexOf(id)
-      if (index >= 0) {
-        forward.splice(index, 1)
-        return session.text('.updated', [id])
-      } else {
-        return session.text('.unchanged', [id])
-      }
-    }).alias('forward.rm')
-
-    register('.clear', async ({ session }) => {
-      session.channel.forward = []
-      return session.text('.updated')
+    cmd.subcommand('.list', {
+      authority: 3,
+      checkArgCount: true,
     })
+      .action(async ({ session }) => {
+        try {
+          const rules = await ctx.database.get('forward', { source: session.cid })
+          session.text('.list', rules)
+        } catch (e) {
+          ctx.logger('forward').error(e)
+          session.text('.error')
+        }
+      }).alias('forward.ls')
 
-    register('.list', async ({ session }) => {
-      const { forward } = session.channel
-      if (!forward.length) return session.text('.empty')
-      return [session.text('.header'), ...forward].join('\n')
-    }).alias('forward.ls')
+    cmd.subcommand('.clear', {
+      authority: 3,
+      checkArgCount: true,
+    })
+      .action(async ({ session }) => {
+        try {
+          await ctx.database.remove('forward', { source: session.cid })
+          session.text('.success')
+        } catch (e) {
+          ctx.logger('forward').error(e)
+          session.text('.error')
+        }
+      })
+    cmd.subcommand('.remove <channel:channel>', {
+      authority: 3,
+      checkArgCount: true,
+    })
+      .option('user', '-U <user:user>')
+      .option('flag', '-F <flag:string>')
+      .action(async ({ session, options }, target) => {
+        try {
+          const res = await ctx.database.get('forward', {
+            $and: [
+              { source: session.cid },
+              { target: target },
+            ],
+          })
+          if (res.length > 1) {
+            session.text('.multi-res')
+            if (!options.flag && !options.user) throw new Error('No filter is mentioned.')
+            else if (options.flag && options.user) throw new Error('More than 1 filters are mentioned.')
+            else {
+              // @ts-ignore
```

Comment:
If `@ts-ignore` appears in your code, it's probably because you wrote something wrong.